server/src/internalClusterTest/java/org/elasticsearch/action/termvectors/GetTermVectorsIT.java

```diff
@@ -415,19 +416,24 @@ public void testDuelESLucene() throws Exception {
         // we generate as many docs as many shards we have
         TestDoc[] testDocs = generateTestDocs("test", testFieldSettings);
 
-        DirectoryReader directoryReader = indexDocsWithLucene(testDocs);
-        TestConfig[] testConfigs = generateTestConfigs(20, testDocs, testFieldSettings);
+        DirectoryReader directoryReader = null;
+        try {
+            directoryReader = indexDocsWithLucene(testDocs);
+            TestConfig[] testConfigs = generateTestConfigs(20, testDocs, testFieldSettings);
+
+            for (TestConfig test : testConfigs) {
+                TermVectorsRequestBuilder request = getRequestForConfig(test);
+                if (test.expectedException != null) {
+                    assertRequestBuilderThrows(request, test.expectedException);
+                    continue;
+                }
 
-        for (TestConfig test : testConfigs) {
-            TermVectorsRequestBuilder request = getRequestForConfig(test);
-            if (test.expectedException != null) {
-                assertRequestBuilderThrows(request, test.expectedException);
-                continue;
+                TermVectorsResponse response = request.get();
+                Fields luceneTermVectors = getTermVectorsFromLucene(directoryReader, test.doc);
+                validateResponse(response, luceneTermVectors, test);
             }
-
-            TermVectorsResponse response = request.get();
-            Fields luceneTermVectors = getTermVectorsFromLucene(directoryReader, test.doc);
-            validateResponse(response, luceneTermVectors, test);
+        } finally {
+            IOUtils.close(directoryReader);
```

Comment:
Failing to close the reader can lead to thread leaks if the randomized searcher enables concurrency.